src/main.rs

```diff
@@ -85,11 +93,16 @@ enum Commands {
         #[clap(long)]
         auth_token: Option<String>,
         /// If this path is provided and it exists, the private key is read from this file and used, if it does not exist the private key will be persisted to this location.
+        ///
+        /// If this path is not provided and persistent is true, the private key will be persisted to the iroh data root.
         #[clap(long)]
         key: Option<PathBuf>,
         /// Optional rpc port, defaults to 4919. Set to 0 to disable RPC.
         #[clap(long, default_value_t = ProviderRpcPort::Enabled(DEFAULT_RPC_PORT))]
         rpc_port: ProviderRpcPort,
+        /// If true, the provider will read and write from the iroh data root to persist data.
+        #[clap(long, default_value = "true")]
+        persistent: Option<bool>,
```

Comment:
Is this really only needed for tests? This complicates the behaviour quite a bit.

Reply:
I think it is nice to have the option to use iroh without touching some common dir.

I remember running `IPFS_PATH=$HOME/.public_ipfs ipfs` and similar things to have ipfs leave my ~/.ipfs alone.

I had briefly considered to split it up so that provide is always just a oneshot thing without persistence, and there is a new `start` subcommand that always uses persistence.

Reply:
But you can do that with `IROH_PATH` as well now, can't you? So `IRO_PATH=/tmp/foo` iroh would do the same temporary thing

Reply:
Sure. The question is if we want to provide an easy way to use iroh provide in a "oneshot" way without any persistence.

Reply:
So the question of persistence here actually centers around the interaction between `key`, identity, and network information. Because we will likely be persisting peer and network information, it would be beneficial for the user to use that information when they provide or fetch, regardless of whether they want to keep the collection they are providing or fetching. 

I think that the story of persisting during fetch and provide will actually come into play when we do garbage collecting / naming / tagging rather than the act of loading up the database itself. But I agree that we will definitely want this "easy path" option.

So I would vote for:
- remove `key` and only use a new identify/persistence directory when the `IROH_PATH` is different
- eventually, have some flag or mechanism for "not persisting" the provided or fetched thing, but for now keep the default of 'always persist'.